### DISCUSSION TITLE
Autoconfigure services with decorators

### DISCUSSION THREAD

**Discussion by sgomez** [ets-labs/python-dependency-injector#548] [Show and tell]
Hi people,

I did a demo to try to autoconfigure services using decorators with a DynamicContainer.

You can add factories automatically with `@service` and command handlers (dict provider) with `@command_handler`.

It works but I'd like to read comments or drawbacks about the suitability of this solution.

https://github.com/sgomez/demo-python-di-with-autoconfiguration

Thanks in advance.

**Comment by rmk135** [comment]
Hi @sgomez ,

Nice work there! I don't see any drawbacks. I personally like to use declarative containers, but it's a question of choice – nothing wrong with using dynamic containers. As a next step, it would be interesting to look at overriding in testing. 

Best,
Roman
